Sources/KSCrashRecording/Monitors/KSCrashMonitor_CPPException.cpp

```diff
@@ -98,14 +102,18 @@ extern "C"
     void __cxa_throw(void* thrown_exception, std::type_info* tinfo, void (*dest)(void*)) __attribute__((disable_tail_calls))
     {
         static cxa_throw_type orig_cxa_throw = NULL;
-        if (g_cxaSwapEnabled == false)
-        {
-            captureStackTrace(NULL, NULL, NULL);
-        }
         unlikely_if(orig_cxa_throw == NULL)
         {
             orig_cxa_throw = (cxa_throw_type) dlsym(RTLD_NEXT, "__cxa_throw");
         }
+        if (tinfo != nullptr && strcmp(tinfo->name(), "NSException") == 0)
+        {
+            return orig_cxa_throw(thrown_exception, tinfo, dest);
+        }
+        if (g_cxaSwapEnabled == false)
+        {
+            captureStackTrace(NULL, NULL, NULL);
+        }
```

Comment:
This `return` is not allowed in this function. 

And I would also suggest just calling `captureStackTrace` with parameters and keep the NSException check there (to also awoid code duplication).
```suggestion
        if (g_cxaSwapEnabled == false)
        {
            captureStackTrace(thrown_exception, tinfo, dest);
        }
```